src/plugins/intel_gpu/src/graph/impls/ocl/crop.cpp

```diff
@@ -41,16 +41,19 @@ struct crop_impl : typed_primitive_impl_ocl<crop> {
 
 public:
     static primitive_impl* create(const crop_node& arg, const kernel_impl_params& impl_param) {
-        const auto& primitive = arg.get_primitive();
-
         auto ew_params = get_default_params<kernel_selector::eltwise_params>(impl_param, 1);
         auto ew_optional_params = get_default_optional_params<kernel_selector::eltwise_optional_params>(arg.get_program());
 
         ew_params.operations.push_back(
             {{kernel_selector::eltwise_params::InputType::Buffer(0)}, kernel_selector::eltwise_mode::ASSIGN});
 
         const auto& input_layout = impl_param.input_layouts[0];
-        ew_params.inputs[0] = convert_data_tensor(input_layout, 1, primitive->offsets);
+        auto desc = impl_param.typed_desc<crop>();
+        if (impl_param.memory_deps.size() > 0) {
+            ew_params.inputs[0] = convert_data_tensor(input_layout, 1, impl_param.input_offsets[0]);
+        } else {
+            ew_params.inputs[0] = convert_data_tensor(input_layout, 1, desc->offsets);
```

Comment:
I think we can merge this, by setting impl_param.input_offsets with desc->offsets in the legacy flow, in the calc_output_layout